src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,33 @@
 package core.basesyntax.service;
 
+import core.basesyntax.RegistrationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null || user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login must be at least " + MIN_LOGIN_LENGTH
+                    + " characters long.");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password must be at least " + MIN_PASSWORD_LENGTH
+                    + " characters long.");
+        }
+        if (user.getAge() == null || user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Age must be at least " + MIN_AGE + ".");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists.");
```

Comment:
Before adding a new user, it is essential to check if the user's login is not null and if the login does not exist in the storage to avoid a potential NullPointerException.